### PR TITLE
fix: Generic assay plugin bugs

### LINKED ISSUE #1982: Generic assay plugin: Inline links in material columns always point to ResultsReports
In this example:
```
Comment[SODAR Assay Link Results]	Standardized Analysis Report
Comment[SODAR Assay Link Row]	Raw Data File;Report File
```

`Raw Data File` columns will link to `Results & Reports` when they should be pointing at the `RowPath`. This behaviour is consistent for all Data and Material headers.

### LINKED ISSUE #1984: Generic assay plugin: Samplesheets cache update fails if RowPath built from ontology column
I want to use the generic assay plugin with a new empty project. ISA-tab upload works fine, the samplesheet table is displayed. After the iRODS collections are created however the samplesheets app does not load anymore and this is in the logs:

```
sodar-celeryd-default_1  | 2024-08-30 14:58:46,807 [ERROR] samplesheets.tasks_celery: Cache update failed for project "Crashtest" (0505fa4a-efdf-4650-97ca-0498042ff64e): sequence item 1: expected str instance, list found
```

### PR DESCRIPTION
**Pull request by sellth**
closes #1982

### CODE REVIEW

**Comment by sellth** [comment]
Now also closes #1984. What happens if we try to create links in an ontology or list column? Should we check for that?

File: samplesheets/assayapps/generic/plugins.py [hunk 1]
```diff
@@ -96,7 +100,13 @@ def _get_col_value(cls, target_col, row, table):
                 header = table['field_header'][i]
                 if header['value'].lower() == target_col.lower():
                     value = row[i]['value']
-        return value
+
+        if isinstance(value, str):
+            return value
+        elif isinstance(value, list) and len(value) == 1:  # OntologyTermRefs
+            return value[0]['name']
+        else:
```
Comment (mikkonie) [inline samplesheets/assayapps/generic/plugins.py]: Here we can leave out the `else:` and simply `return None`, as reaching any of the previous clauses returns from the function anyway.

**Review by mikkonie — CHANGES_REQUESTED** [review]
One small note, otherwise this looks good to merge.

**Comment by mikkonie** [comment]
> Now also closes #1984. What happens if we try to create links in an ontology or list column? Should we check for that?

I would say these cases should be checked, as long as there's a somewhat-realistic chance someone woult attempt to create links to such columns. I'm not sure if it needs to go in this PR though or if it should be considered an additional feature. You make the call :)

**Comment by sellth** [comment]
The project might end up in a state which the user cannot recover from without admin help, but it will probably be an edge case. I would prioritise getting these fixes out.

**Review by mikkonie — APPROVED** [review]

**Pull request merged**